src/ascii_char.rs

```diff
@@ -297,165 +288,323 @@ impl AsciiChar {
+    /// assert_eq!(AsciiChar::new('\t').is_ascii_control(), true);
+    /// assert_eq!(AsciiChar::EOT.is_ascii_control(), true);
     /// ```
     #[inline]
-    pub fn is_control(self) -> bool {
-        self < AsciiChar::Space || self == AsciiChar::DEL
+    pub const fn is_ascii_control(&self) -> bool {
+        ((*self as u8) < b' ') | (*self as u8 == 127)
     }
 
     /// Checks if the character is printable (except space)
     ///
     /// # Examples
     /// ```
-    /// use ascii::ToAsciiChar;
-    /// assert_eq!('n'.to_ascii_char().unwrap().is_graph(), true);
-    /// assert_eq!(' '.to_ascii_char().unwrap().is_graph(), false);
-    /// assert_eq!('\n'.to_ascii_char().unwrap().is_graph(), false);
+    /// # use ascii::AsciiChar;
+    /// assert_eq!(AsciiChar::new('n').is_ascii_graphic(), true);
+    /// assert_eq!(AsciiChar::new(' ').is_ascii_graphic(), false);
+    /// assert_eq!(AsciiChar::new('\n').is_ascii_graphic(), false);
     /// ```
     #[inline]
-    pub fn is_graph(self) -> bool {
+    pub const fn is_ascii_graphic(&self) -> bool {
         self.as_byte().wrapping_sub(b' ' + 1) < 0x5E
     }
 
     /// Checks if the character is printable (including space)
     ///
     /// # Examples
     /// ```
-    /// use ascii::ToAsciiChar;
-    /// assert_eq!('n'.to_ascii_char().unwrap().is_print(), true);
-    /// assert_eq!(' '.to_ascii_char().unwrap().is_print(), true);
-    /// assert_eq!('\n'.to_ascii_char().unwrap().is_print(), false);
+    /// # use ascii::AsciiChar;
+    /// assert_eq!(AsciiChar::new('n').is_ascii_printable(), true);
+    /// assert_eq!(AsciiChar::new(' ').is_ascii_printable(), true);
+    /// assert_eq!(AsciiChar::new('\n').is_ascii_printable(), false);
     /// ```
     #[inline]
-    pub fn is_print(self) -> bool {
+    pub const fn is_ascii_printable(&self) -> bool {
         self.as_byte().wrapping_sub(b' ') < 0x5F
     }
 
-    /// Checks if the character is alphabetic and lowercase
+    /// Checks if the character is alphabetic and lowercase (a-z).
     ///
     /// # Examples
     /// ```
-    /// use ascii::ToAsciiChar;
-    /// assert_eq!('a'.to_ascii_char().unwrap().is_lowercase(), true);
-    /// assert_eq!('A'.to_ascii_char().unwrap().is_lowercase(), false);
-    /// assert_eq!('@'.to_ascii_char().unwrap().is_lowercase(), false);
+    /// use ascii::AsciiChar;
+    /// assert_eq!(AsciiChar::new('a').is_lowercase(), true);
+    /// assert_eq!(AsciiChar::new('A').is_lowercase(), false);
+    /// assert_eq!(AsciiChar::new('@').is_lowercase(), false);
     /// ```
     #[inline]
-    pub fn is_lowercase(self) -> bool {
+    pub const fn is_lowercase(self) -> bool {
         self.as_byte().wrapping_sub(b'a') < 26
     }
 
-    /// Checks if the character is alphabetic and uppercase
+    /// Checks if the character is alphabetic and lowercase (a-z).
+    ///
+    /// This method is identical to [`is_lowercase()`](#method.is_lowercase)
+    pub fn is_ascii_lowercase(&self) -> bool {
```

Comment:
Same here, could we make this `const` and just drop the `is_lowercase` method? I don't think it's worth duplicating just for the difference in self.

Reply:
`char` has both.